src/fastoad/models/performances/mission/polar.py

```diff
@@ -28,17 +44,80 @@ def __init__(self, cl: ndarray, cd: ndarray):
 
         Once defined, for any CL value, CD can be obtained using :meth:`cd`.
 
-        :param cl: a N-elements array with CL values
-        :param cd: a N-elements array with CD values that match CL
+        :param input_arg: a dictionary containing the CL and CD N-elements
+        array and additional variables for ground effect calculation.
+        For ground segments the 'CL vs alpha curve' is required and the dictionary
+        must contain 'CL0_clean', 'CL_alpha' and 'CL_high_lift'
+
         """
-        self._definition_CL = cl
-        self._cd = interp1d(cl, cd, kind="quadratic", fill_value="extrapolate")
+        if isinstance(input_arg, tuple):
+            self._definition_CL = input_arg[0]
+            self._cd = interp1d(
+                input_arg[0], input_arg[1], kind="quadratic", fill_value="extrapolate"
+            )
+        elif isinstance(input_arg, dict):
+            key = list(input_arg.keys())
+            if "CL" and "CD" in key:
+                self._definition_CL = input_arg["CL"]
+                self._cd = interp1d(
+                    input_arg["CL"], input_arg["CD"], kind="quadratic", fill_value="extrapolate"
+                )
+                if "CL0_clean" and "CL_alpha" and "CL_high_lift" in key:
+                    self.init_takeoff_polar(input_arg)
+                    self._use_CL_alpha = True
+                if "ground_effect" in key:
+                    self.init_ground_effect(input_arg)
+        else:
+            # default initialisation for instanciation without argument
+            self._definition_CL = np.array([0.2, 0.5, 1.0])
+            self._definition_CD = np.array([0.01, 0.02, 0.1])
+
+            self._cd = interp1d(
+                self._definition_CL, self._definition_CD, kind="quadratic", fill_value="extrapolate"
+            )
 
         def _negated_lift_drag_ratio(lift_coeff):
             """Returns -CL/CD."""
             return -lift_coeff / self.cd(lift_coeff)
 
-        self._optimal_CL = fmin(_negated_lift_drag_ratio, cl[0], disp=0)
+        self._optimal_CL = fmin(_negated_lift_drag_ratio, self._definition_CL[0], disp=0)
+
+    def init_ground_effect(self, input_arg):
+        """
+        Initialise the ground effect calculation
+
+        Only Raymer model available
+        """
+        if input_arg["ground_effect"] == "Raymer":
+            # TO DO : replace condition by keywords designating the ground effect model
+            self._span = input_arg["span"]
+            self._lg_height = input_arg["lg_height"]
+            self._induced_drag_coef = input_arg["induced_drag_coef"]
+            self._k_winglet = input_arg["k_winglet"]
+            self._k_cd = input_arg["k_cd"]
```

Comment:
I understand why you did not want to have all these variables as inputs along already existing ones.

But the correct way to manage that is to create a class dedicated to ground effect. And an instance of this class should be expected in the `__init__` of the polar.

This looks even more a good approach since you want to allow the choice between several ways of computing the ground effect.

Reply:
Implemented in 36bab9f464f2e077bc2e86f938d1c440b7c47158